### PR TITLE
Split main workflow into build, lint and test workflows

### PR DESCRIPTION
**Pull request by buberdds**

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  # Or when a pull request event occurs for a pull request against one of the
+  # matched branches.
+  pull_request:
+    branches: [stable, master]
+
+# disable secrets.GITHUB_TOKEN permissions
+permissions: {}
+
+jobs:
+  yarn_cache:
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:18]: Do we need this `yarn_cache` job here or just in the `ci-test` workflow?

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Trigger the workflow when:
+on:
+  # A push occurs to one of the matched branches.
+  push:
+    branches: [stable, master]
+  # Or when a pull request event occurs for a pull request against one of the
+  # matched branches.
+  pull_request:
+    branches: [stable, master]
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:12]:
```yaml
# Trigger the workflow when:
on:
  # A push occurs to one of the matched branches.
  push:
    branches:
      - master
      - stable/*
  # Or when a pull request event occurs for a pull request against one of the
  # matched branches.
  pull_request:
    branches:
      - master
      - stable/*
```
We name stable branches with `stable/` prefix, so we should use something like in Oasis Core.

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# disable secrets.GITHUB_TOKEN permissions
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:14]:
```
# Explicitly disable secrets.GITHUB_TOKEN permissions.
```

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+
+  build:
+    # NOTE: This name appears in GitHub's Checks API.
+    name: build
+    needs: [yarn_cache]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:37]: Use `actions/checkout@v4` and add `name:` `Checkout code` to unify this with the https://github.com/oasisprotocol/explorer/blob/master/.github/workflows/ci-build.yml flow.

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Use Node.js 18.x
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:38]: Unify terminology with https://github.com/oasisprotocol/explorer/blob/master/.github/workflows/ci-build.yml (i.e. `Set up Node.js 18`).

File: .github/workflows/ci-build.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: 'yarn'
```
Comment (tjanez) [inline .github/workflows/ci-build.yml:42]: Omit `'`s

File: .github/workflows/ci-lint.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# NOTE: This name appears in GitHub's Checks API and in workflow's status badge.
```
Comment (tjanez) [inline .github/workflows/ci-lint.yml:1]: Same comments apply as for `ci-build.yml`.

File: .github/workflows/ci-test.yml [hunk 8]
```diff
@@ -1,9 +1,13 @@
-name: Build and test
+# NOTE: This name appears in GitHub's Checks API and in workflow's status badge.
```
Comment (tjanez) [inline .github/workflows/ci-test.yml:1]: Same comments apply as for `ci-build.yml`.

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (buberdds) [reply .github/workflows/ci-build.yml]: We don't

File: README.md [hunk 9]
```diff
@@ -182,9 +183,10 @@ Transifex a few hours after changes are merged. After they are translated, click
 _"i18n: Update translations from Transifex"_.
 
 Adding a new language:
+
 1. first add it to Transifex and translate the strings,
 2. create a folder with the new language code in `src/locales` and download the
-  translation file there,
+    translation file there,
```
Comment (buberdds) [inline README.md:189]: updated due to MegaLinter `Wrong amount of left-padding spaces(want multiple of 2)`

**Review by tjanez — APPROVED** [review]
Looks good, thanks @buberdds!

**Pull request merged**